x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java

```diff
@@ -84,6 +85,17 @@ public class RemoteClusterSecurityRestIT extends AbstractRemoteClusterSecurityTe
             .build();
     }
 
+    /**
+     * This is needed because `randomBoolean` is not available inside the outer `static {}` scope,
+     * where the initial cluster settings are set
+     */
+    private static String atomicallyRandomizeSslEnabled(AtomicReference<Boolean> sslEnabledRef) {
+        if (sslEnabledRef.get() == null) {
+            sslEnabledRef.set(randomBoolean());
+        }
+        return String.valueOf(sslEnabledRef.get());
+    }
+
     @ClassRule
     // Use a RuleChain to ensure that fulfilling cluster is started before query cluster
     public static TestRule clusterRule = RuleChain.outerRule(fulfillingCluster).around(queryCluster);
```

Comment:
Alternatively, we can have something like
```suggestion
    public static TestRule clusterRule = RuleChain.outerRule(new RunnableTestRuleAdapter(() -> sslEnabled.set(usually())))
        .around(fulfillingCluster)
        .around(queryCluster);
```

Note I also use `usually` instead of `randomBoolean` to bias towards SSL enabled test scenarios.

Reply:
Super neat, TIL.